src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/XmlResolverHelper.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using System.Threading.Tasks;
+using System.Xml;
+
+namespace System.Security.Cryptography.Xml
+{
+    internal static class XmlResolverHelper
+    {
+        public static XmlResolver GetThrowingResolver()
+        {
+#if NET7_0_OR_GREATER
+            return XmlResolver.ThrowingResolver;
+#else
+            return XmlThrowingResolver.Singleton;
+#endif
+        }
+
+#if !NET7_0_OR_GREATER
+        // An XmlResolver that forbids all external entity resolution.
+        // (Copied from XmlResolver.ThrowingResolver.cs.)
```

Comment:
could the visibility of throwing resolver just be internal for `#if !NET7_0_OR_GREATER`? Seems it would remove all `#if`s here

Reply:
I missed it's in different project. Still you could do the same but add conditional in csproj to include that file, possibly something similar to what we do with attributes in System.Private.CoreLib:
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/RequiredMemberAttribute.cs#L8

Reply:
I agree it's weird, but it is a pattern we've done before in a few places. See for instance https://github.com/dotnet/runtime/blob/7be37908e5a1cbb83b1062768c1649827eeaceaa/src/libraries/Common/src/System/Security/Cryptography/CryptoThrowHelper.Windows.cs and https://github.com/dotnet/runtime/blob/7be37908e5a1cbb83b1062768c1649827eeaceaa/src/libraries/Common/src/System/ThrowHelper.cs.